docs/v1/guide/best-practices.md

```diff
@@ -0,0 +1,614 @@
+const video = document.querySelector('video')
+// Check if the video exists and is ready
+if (video && video.readyState > 0) {
+  // Get the title of the video
+  const title = document.querySelector('.video-title')?.textContent || 'Unknown video'
+  // Get the author of the video
+  const author = document.querySelector('.video-author')?.textContent || 'Unknown author'
+  // Check if the video is playing
+  const isPlaying = !video.paused
+  // Set the activity type to watching
+  presenceData.type = ActivityType.Watching
+  // Set the details to the title
+  presenceData.details = title
+  // Set the state to the author
+  presenceData.state = `By ${author}`
+}
+```
+
+### Follow the Style Guide
+
+Use consistent formatting and naming conventions. This makes your code easier to read and maintain.
+
+```typescript
+// Good
+const presence = new Presence({
+  clientId: 'your_client_id'
+})
+
+const SLIDESHOW_TIMEOUT = 5000
+
+function getVideoInfo() {
+  const video = document.querySelector('video')
+  return {
+    title: document.querySelector('.video-title')?.textContent || 'Unknown video',
+    author: document.querySelector('.video-author')?.textContent || 'Unknown author',
+    isPlaying: video ? !video.paused : false
+  }
+}
+
+// Bad
+const presence = new Presence({ clientId: 'your_client_id' })
+const slideshow_timeout = 5000
+function get_video_info() {
+  const video = document.querySelector('video')
+  return { title: document.querySelector('.video-title')?.textContent || 'Unknown video', author: document.querySelector('.video-author')?.textContent || 'Unknown author', isPlaying: video ? !video.paused : false }
+}
+```
+
+## Performance
+
+### Minimize DOM Queries
+
+Cache DOM elements that you use multiple times to avoid unnecessary DOM queries.
+
+```typescript
+// Good
+const video = document.querySelector('video')
+const title = document.querySelector('.video-title')?.textContent || 'Unknown video'
+const author = document.querySelector('.video-author')?.textContent || 'Unknown author'
+
+if (video && video.readyState > 0) {
+  presenceData.details = title
+  presenceData.state = `By ${author}`
+
+  if (!video.paused) {
+    // Use video
+  }
+}
+
+// Bad
+if (document.querySelector('video') && document.querySelector('video').readyState > 0) {
+  presenceData.details = document.querySelector('.video-title')?.textContent || 'Unknown video'
+  presenceData.state = `By ${document.querySelector('.video-author')?.textContent || 'Unknown author'}`
+
+  if (!document.querySelector('video').paused) {
+    // Use video
+  }
+}
+```
+
+### Use Efficient Selectors
+
+Use efficient selectors to find elements on the page. IDs are the fastest, followed by classes, then tag names.
+
+```typescript
+// Good
+const video = document.getElementById('video')
+const title = document.querySelector('.video-title')
+
+// Bad
+const video = document.querySelector('video')
+const title = document.getElementsByClassName('video-title')[0]
```

Comment:
const video = document.querySelector('video') is actually a good selector as there is usually only 1 video playing on a page.